include/bluetooth/audio/cap.h

```diff
@@ -70,6 +70,57 @@ struct bt_cap_cb {
 	 */
 	void (*ccid_found)(struct bt_conn *conn, int err, uint8_t ccid,
 			   const struct bt_cap_ccid_record *ccid_record);
+
+#if defined(CONFIG_BT_CAP_INITIATOR) && defined(CONFIG_BT_AUDIO_UNICAST_CLIENT)
+	/**
+	 * @brief Callback for bt_cap_unicast_discover()
+	 *
+	 * @param conn The connection pointer supplied to
+	 *             bt_cap_unicast_discover()
+	 * @param err  0 if Common Audio Service was found, else errno value if
+	 *             negative or ATT error code if positive.
+	 */
+	void (*discovery_complete)(struct bt_conn *conn, int err);
+
+	/**
+	 * @brief Callback for bt_cap_unicast_audio_start()
+	 *
+	 * @param unicast_group The unicast group pointer supplied to
+	 *                       bt_cap_unicast_audio_start()
+	 * @param err            0 if success, else errno value if
+	 *                       negative or ATT error code if positive.
+	 * @param conn           Pointer to the connection where the error
+	 *                       occurred. NULL if @p err is 0.
+	 */
+	void (*unicast_start_complete)(struct bt_audio_unicast_group *unicast_group,
+				       int err, struct bt_conn *conn);
```

Comment:
is this callback called multiple times per each member in the group CAP is starting?

Reply:
Hmm, the idea was to only call it once for either success (for all devices) or failure (for any device). Do you think it would be better to periodically update the application with the status?

Reply:
I'm asking because there is `struct bt_conn *conn` parameter which makes me think it is per device. Calling it once should be enough but bt_conn is not needed, as it should be a list of streams inside the group. Is that correct?

Reply:
The `conn` parameter is supplied here to provide information about which device was the cause of the failure, as per the documentation:
```
	 * @param conn           Pointer to the connection where the error
	 *                       occurred. NULL if @p err is 0.
```

I figured it would be useful for the application to know which device caused the issue, e.g. if that specific device disconnected during this procedure, so that the application can perform a reconnect or remove it from the set and restart the procedure.